### PR TITLE
Add pparam to url_sig plugin to make it authenticate pristine URL, eliminate scheme check, other minor changes.

### PR DESCRIPTION
**Pull request by ywkaras**

### CODE REVIEW

**Comment by ywkaras** [comment]
Why is the apply-named RAT CI test deciding it should run my shell script and then not liking the results on absolutely no basis whatsoever?

**Comment by bryancall** [comment]
You need to put licensing information on tests/gold_tests/pluginTest/url_sig/run_sign.sh

**Comment by bryancall** [comment]
[approve ci ubuntu]

**Comment by ywkaras** [comment]
[ approve ci ubuntu ]

**Comment by ywkaras** [comment]
[approve ci ubuntu]

File: plugins/experimental/url_sig/url_sig.c [hunk 1]
```diff
@@ -422,35 +443,34 @@ TSRemapDoRemap(void *ih, TSHttpTxn txnp, TSRemapRequestInfo *rri)
...
   }
   // Expiration
-  p = strstr(query, EXP_QSTRING "=");
-  if (p != NULL) {
-    p += strlen(EXP_QSTRING) + 1;
-    expiration = atoi(p);
-    if (expiration == 0 || expiration < time(NULL)) {
+  cp = strstr(query, EXP_QSTRING "=");
+  if (cp != NULL) {
+    cp += strlen(EXP_QSTRING) + 1;
```
Comment (SolidWallOfCode) [inline plugins/experimental/url_sig/url_sig.c]: `sizeof(EXP_QSTRING)-1` vs. `strlen`?

File: plugins/experimental/url_sig/url_sig.c [hunk 2]
```diff
@@ -422,35 +443,34 @@ TSRemapDoRemap(void *ih, TSHttpTxn txnp, TSRemapRequestInfo *rri)
...
     goto deny;
   }
   // Algorithm
-  p = strstr(query, ALG_QSTRING "=");
-  if (p != NULL) {
-    p += strlen(ALG_QSTRING) + 1;
-    algorithm = atoi(p);
+  cp = strstr(query, ALG_QSTRING "=");
+  if (cp != NULL) {
+    cp += strlen(ALG_QSTRING) + 1;
```
Comment (SolidWallOfCode) [inline plugins/experimental/url_sig/url_sig.c]: `sizeof` vs. `strlen`?

File: plugins/experimental/url_sig/url_sig.c [hunk 3]
```diff
@@ -570,7 +606,7 @@ TSRemapDoRemap(void *ih, TSHttpTxn txnp, TSRemapRequestInfo *rri)
     rri->redirect = 1;
     break;
   default:
-    TSHttpTxnErrorBodySet(txnp, TSstrdup("Authorization Denied"), strlen("Authorization Denied") - 1, TSstrdup("text/plain"));
+    TSHttpTxnErrorBodySet(txnp, TSstrdup("Authorization Denied"), sizeof("Authorization Denied") - 1, TSstrdup("text/plain"));
```
Comment (SolidWallOfCode) [inline plugins/experimental/url_sig/url_sig.c]: Why use `sizeof` here and `strlen` on `EXP_QSTRING` and `ALG_QSTRING` earlier?

**Comment by SolidWallOfCode** [comment]
Overall it looks good to me, just one trivial issue.

File: plugins/experimental/url_sig/url_sig.c (hunk 3 above)
Comment (ywkaras) [reply plugins/experimental/url_sig/url_sig.c]: Because I'm not a good person.

**Pull request merged**